trie/zk_merkle_trie.go

```diff
@@ -12,11 +12,36 @@ import (
 
 type ZkMerkleTrie struct {
 	*zk.MerkleTree
-	db *Database
+	db        *Database
+	mutateKey func(key []byte) []byte
```

Comment:
Why don't we rename this? Such as `transformKey`, `preProcessKey`, or `reverseKey`.

Reply:
i'll change it to a `transform` key.